File: examples/python/basic_tracking.py
"""Basic cost tracking with OpenAI."""

from compute_cfo import wrap
from openai import OpenAI

# Wrap your client — one line, zero config
client = wrap(OpenAI())

response = client.chat.completions.create(
    model="gpt-4o",
    messages=[{"role": "user", "content": "What is the capital of France?"}],
    metadata={"project": "demo", "agent": "qa-bot"},
)

print(response.choices[0].message.content)

File: examples/typescript/basic_tracking.ts
/**
 * Basic cost tracking with OpenAI.
 */

import { wrap } from 'compute-cfo';
import OpenAI from 'openai';

const client = wrap(new OpenAI());

const response = await client.chat.completions.create({
  model: 'gpt-4o',
  messages: [{ role: 'user', content: 'What is the capital of France?' }],
  metadata: { project: 'demo', agent: 'qa-bot' },
});

console.log(response.choices[0].message.content);
